.github/actions/check-pr/index.ts

```diff
@@ -0,0 +1,120 @@
+import { setOutput, getInput, setFailed, info, notice } from '@actions/core';
+import { context } from '@actions/github';
+import { readFile } from 'node:fs/promises';
+import { resolve } from 'node:path';
+
+const validPreambles = ['feat', 'fix', 'chore'];
+interface ParsedTitle {
+  preamble?: string;
+  topic?: string;
+  isBreaking: boolean;
+  description?: string;
+}
+
+function parseTitle(title: string): ParsedTitle {
+  const groups = title.match(
+    /(?<preamble>\w+)(\((?<topic>\w+)\))?(?<isBreaking>!)?: (?<description>.*)/
+  ).groups;
+  if (groups) {
+    return { ...groups, isBreaking: !!groups.isBreaking };
+  } else {
+    throw new Error('Could not parse PR title');
```

Comment:
not an regex expert, but could we print more information to help understand what's wrong?

Reply:
I changed this quite a bit, I hope it is more readable now.